fmn/consumer/digests.py

```diff
@@ -0,0 +1,134 @@
+# FMN worker calculating and sending digests
+
+
+import json
+import logging
+import time
+import random
+
+import pika
+import fedmsg
+import fedmsg.meta
+import fedmsg_meta_fedora_infrastructure
+
+from pika.adapters import twisted_connection
+from twisted.internet import defer, reactor, protocol,task
+
+import fmn.lib
+import fmn.rules.utils
+import fmn.consumer.backends as fmn_backends
+import fmn.consumer.producer as fmn_producers
+
+import fmn.consumer.fmn_fasshim
+from fedmsg_meta_fedora_infrastructure import fasshim
+
+log = logging.getLogger("fmn")
+log.setLevel('DEBUG')
+CONFIG = fedmsg.config.load_config()
+fedmsg.meta.make_processors(**CONFIG)
+
+DB_URI = CONFIG.get('fmn.sqlalchemy.uri', None)
+session = fmn.lib.models.init(DB_URI)
+
+fmn.consumer.fmn_fasshim.make_fas_cache(**CONFIG)
+# Monkey patch fedmsg_meta modules
+fasshim.nick2fas = fmn.consumer.fmn_fasshim.nick2fas
+fasshim.email2fas = fmn.consumer.fmn_fasshim.email2fas
+fedmsg_meta_fedora_infrastructure.supybot.nick2fas = \
+    fmn.consumer.fmn_fasshim.nick2fas
+fedmsg_meta_fedora_infrastructure.anitya.email2fas = \
+    fmn.consumer.fmn_fasshim.email2fas
+fedmsg_meta_fedora_infrastructure.bz.email2fas = \
+    fmn.consumer.fmn_fasshim.email2fas
+fedmsg_meta_fedora_infrastructure.mailman3.email2fas = \
+    fmn.consumer.fmn_fasshim.email2fas
+fedmsg_meta_fedora_infrastructure.pagure.email2fas = \
+    fmn.consumer.fmn_fasshim.email2fas
+
+CNT = 0
+
+log.debug("Instantiating FMN digest producer")
+backend_kwargs = dict(config=CONFIG)
+
+
+OPTS = pika.ConnectionParameters(
+    heartbeat_interval=0,
+    retry_delay=2,
+)
+connection = pika.BlockingConnection(OPTS)
+
+
+
+class FakeBackend(object):
+    def inform_workers(self, body):
+        queue = 'backends'
+        chan = self.connection.channel()
+        chan.exchange_declare(exchange=queue)
+        chan.queue_declare(queue, durable=True)
+
+        body['backend'] = self.name
+
+        chan.basic_publish(
+            exchange='',
+            routing_key=queue,
+            body=json.dumps(body),
+            properties=pika.BasicProperties(
+                delivery_mode=2
+            )
+        )
+        chan.close()
+
+    def __init__(self, name, connection):
+        self.name = name
+        self.connection = connection
+
+    def handle(self, session, recipient, msg, streamline=False):
+        self.inform_workers({
+            'method': 'handle',
+            'recipient': recipient,
+            'msg': msg,
+            'streamline': steamline})
+
+    def handle_batch(self, session, recipient, queued_messages):
+        self.inform_workers({
+            'method': 'handle_batch',
+            'recipient': recipient,
+            'queued_messages': queued_messages})
+
+# If debug is enabled, use the debug backend everywhere
```

Comment:
This comment implies an if statement for whether debug is enabled, but I don't see an if statement. Actually, a lot about this entire module seems like it's for debugging, but the name of the module doesn't sound like that. Is that what this is for?